pkg/kine/prepared/db.go

```diff
@@ -0,0 +1,191 @@
+package prepared
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"sync"
+	"sync/atomic"
+)
+
+type DB struct {
+	underlying *sql.DB
+	mu         sync.Mutex
+	maxSize    int
+	store      map[string]*lruEntry
```

Comment:
It's not immediately clear to me what this is a store of, also if we add more similar stores in future we'll have a name-clash. Could this be renamed something like `preparedQueryCache`?

Also, this cache functionality, although strongly related to the `DB`, I wonder whether it ought to go in its own separate struct? I think it could be really nice and clean to have calls like `db.cache.get()` and `db.cache.put()`

Reply:
As discussed on MM, I will remove the LRU part.